onTruckPlugin/src/main/java/com/github/ontruck/AutonomousController.java

```diff
@@ -0,0 +1,135 @@
+package com.github.ontruck;
+
+public class AutonomousController extends Thread {
+
+	private final DistanceSensor sensor;
+	private final PlanExecutor executor;
+
+	//TODO: here we need the distance to the object in front
+	private int currentDistance;
+	private final int GOAL_DISTANCE = 40;
+	private final byte MAX_POWER = 20;
+	private final long loopDelay = 10;
+	private long latestSensorTimeStamp = 0;
+	private boolean haveJumpedOneSensorBatch = false;
+
+	public AutonomousController(DistanceSensor sensor, PlanExecutor executor) {
+		this.sensor = sensor;
+		currentDistance = sensor.getLatestFilteredDistance().getY();
+		this.executor = executor;
+	}
+
+	//Returns false if speed does not need to be altered, return true if speed has been altered.
+	private Plan generatePlan() {
+
+		// Accepted difference between current distance and goal distance (to prevent constant acceleration, this depends on the accuracy of the distance measurement)
+		int distanceMargin = 20;
+
+		//How much distance must have changed since last measurement for us to care
+		int relativeVelocity = relativeVelocity();
+
+		//System.out.printf("Δd: [%4d]", relativeVelocity);
+
+		int relativeVelocityLimit = 2;
+
+		//current distance may never be smaller than goal distance, but it can be greater to the margin.
+		if (currentDistance >= GOAL_DISTANCE && currentDistance < GOAL_DISTANCE + distanceMargin) {
+			if (relativeVelocity < -relativeVelocityLimit) { //If distance was just decreased a lot, decrease speed
+				return new Plan(
+						new Instruction(
+								Instruction.InstructionType.DecreaseSpeed,
+								MAX_POWER
+						)
+				);
+			} else if (relativeVelocity > relativeVelocityLimit) { //If distance was just increased a lot, increase speed
+				return new Plan(
+						new Instruction(
+								Instruction.InstructionType.IncreaseSpeed,
+								MAX_POWER
+						)
+				);
+			} else { //Distance was good and has not just changed a lot, is good
+				return new Plan();
+			}
+			//currentDistance is greater than GOAL_DISTANCE
+		} else if (currentDistance > GOAL_DISTANCE + distanceMargin) {
+			//If difference is significantly negative, we do not need to accelerate more
+			if (relativeVelocity < -relativeVelocityLimit) {
+				return new Plan();
+			} else {
+				return new Plan(
+						new Instruction(
+								Instruction.InstructionType.IncreaseSpeed,
+								MAX_POWER
+						)
+				);
+			}
+		} else if (currentDistance < GOAL_DISTANCE) { //currentDistance is smaller than GOAL_DISTANCE
+			if (relativeVelocity > relativeVelocityLimit) {
+				return new Plan();
+			} else {
+				return new Plan(
+						new Instruction(
+							Instruction.InstructionType.Drive,
+							(byte)0
+						)
+				);
+			}
+		}
+		// Do nothing
+		return new Plan();
+	}
+
+	//This function calculates the velocity relative to the followed object over the last five measures
+	private int relativeVelocity() {
```

Comment:
This method is changed to no longer calculate a relative velocity, but rather just a change in distance. The name of the method is thus misgiving and should be refactored. Also, maybe remove the commented code that is not used any more, and comment the method to show what it does.